aion.sh

```diff
@@ -37,5 +37,169 @@ ARG=$@
 # add execute permission to rt
 chmod +x ./rt/bin/*
 
-env EVMJIT="-cache=1" ./rt/bin/java -Xms4g \
-        -cp "./lib/*:./lib/libminiupnp/*:./mod/*" org.aion.Aion "$@"
+#env EVMJIT="-cache=1" ./rt/bin/java -Xms4g \
+#	-cp "./lib/*:./lib/libminiupnp/*:./mod/*" org.aion.Aion "$@" &
+
+
+####### WATCHGUARD IMPLEMENTATION #######
+#				    	#
+#   To kill: ps aux | egrep "aion.sh" 	#
+#					#
+#   Wait - reboot timer condition	#
+#   Sample - rebounce checking rate	#
+#   Tolerance - kernel dead condition	#
+#   ThreadRate - frequency of checking	#
+#	thread dead condition (sample)  #
+#					#
+#########################################
+
+# Enable "watch" as first command line argument
+guard=false
+if [[ $1 == "watch" ]]; then
+  first=true
+  for arg in $@; do
+    if $first; then
+      set --
+      first=false
+    else		  
+      set -- "$@" "$arg"
+    fi
+  done
+  guard=true
+fi
+
+if $guard; then
+
+	wait=300	# sec
+	sample=30	# sec
+	tolerance=60 	# sec
+	threadRate=3 	# rate
+	#wait=30
+	#sample=1
+	#tolerance=1
+	#threadRate=20
+
+	config=config/config.xml
+	logging=$(egrep -o "log-file.*log-file" $config | cut -d">" -f2 | cut -d"<" -f1)
+	logpath=$(egrep -o "log-path.*log-path" $config | cut -d">" -f2 | cut -d"<" -f1)
+	file=$logpath/aionCurrentLog.dat
+
+	noInterrupt=true
+	countRebounce=0
+	running=false
+	watching=false
+	lastBoot=0
+
+	trap "exit" INT TERM
+	trap "interrupt" EXIT
+	function interrupt() {
+
+		# Interrupts the Aion kernel and awaits shutdown complete
+		if $running; then
+		  kill $kPID
+		  temp=$(top -n1 -p $kPID | egrep -o "$kPID")
```

Comment:
Minor suggestion: I would do something like "ps --pid $kPID" instead of "top -n1 -p $kPID" (and everywhere else you use top)

The top command prints a bunch of memory info and other stuff that might get accidentally picked up by the egrep.